pallets/msa/src/lib.rs

```diff
@@ -213,6 +236,20 @@ impl<T: Config> Pallet<T> {
 	pub fn get_owner_of(key: &T::AccountId) -> Option<MessageSenderId> {
 		Self::get_key_info(&key).map(|info| info.msa_id)
 	}
+
+	/// Fetches all the keys associated with a message Source Account
+	/// NOTE: This should only be called from RPC due to heavy database reads
```

Comment:
This Note message makes me think that we are not doing a good job at compartmentalizing logic.

Reply:
what is the alternative?

I can pull out the logic from here inside RPC but having all the logic inside one place is nice but don't have any strong opinions on it.